### PR TITLE
feat(ivy): ngDevMode has state getter methods

### PR DESCRIPTION
**Pull request by benlesh**
- Moves all perf counters under `ngDevMode.perfCounters`
- Adds all global state getter methods to `ngDevMode.state`. For example: `ngDevMode.state.getLView()` or `ngDevMode.state.getSelectedIndex()`.

This should be useful when debugging issues with global state.

### CODE REVIEW

File: packages/core/test/acceptance/listener_spec.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import {By} from '@angular/platform-browser';
 import {onlyInIvy} from '@angular/private/testing';
 
 function getNoOfNativeListeners(): number {
-  return ngDevMode ? ngDevMode.rendererAddEventListener : 0;
+  return ngDevMode ? ngDevMode.perfCounters.rendererAddEventListener : 0;
```
Comment (ocombe) [inline packages/core/test/acceptance/listener_spec.ts]:
```suggestion
  return ngDevMode && ngDevMode.perfCounters? ngDevMode.perfCounters.rendererAddEventListener : 0;
```

**Review by mhevery — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
@vikerman is there anything on g3 side that needs to be updated to support this change (based on your work to add `ngDevMode` build time flag)?

**Review by vikerman — APPROVED** [review]
Can't see anything that will be problematic from the outset. Have the change to remove ngDevMode inside google3. A TAP presubmit should be able to catch any issues with that

**Comment by benlesh** [comment]
I think I have a better idea for this I'm going to try out in a different PR... I'm marking this as blocked until then...

**Pull request closed without merging**